src/planner/binder/statement/bind_create.cpp

```diff
@@ -54,7 +54,8 @@ void Binder::BindSchemaOrCatalog(ClientContext &context, string &catalog, string
 		if (database) {
 			// we have a database with this name
 			// check if there is a schema
-			auto schema_obj = Catalog::GetSchema(context, INVALID_CATALOG, schema, OnEntryNotFound::RETURN_NULL);
+			auto schema_obj = Catalog::GetSchema(context, INVALID_CATALOG, schema, OnEntryNotFound::RETURN_NULL,
```

Comment:
Maybe instead of adding an extra parameter to `GetSchema` we can push a virtual function into the catalog that does this, e.g.:

```sql
catalog.CheckAmbiguousCatalogOrSchema(context, schema);
```

That can then have the default implementation of running `GetSchema` and checking if it exists - but can be overriden.

That's a bit more targeted and clear in what the purpose is, plus we can avoid breaking existing code that overrides `GetSchema`.

Reply:
Thanks. Overriding GetSchema was not possible yet (this PR added virtual for that) but your suggestion is indeed more targeted, and works just as well for MotherDuck. Can rework it.

Reply:
so.. while implementing this new method I was porting the stuff using the search path and it kinda dawned on me that the search_path API already allows to ask which catalogs exist that have a schema with a certain name. These were already used by the GetCatalogEntries() method in catalog.cpp and interestingly provide the same info without opening a transaction (fine with me!!). 

Anyways, it is late, maybe I am getting this wrong, but it seems there is no need to create new (virtual) methods in catalog. Just move the implementation of the ambiguity check from catalog API to search_path API

Seems to work..
<img width="806" alt="Screenshot 2024-02-22 at 01 20 54" src="https://github.com/duckdb/duckdb/assets/9865047/985eccc5-008f-44d0-9d12-2bfc24537c59">

Reply:
OK, after discord conversation with Mark I did the CheckAmbiguousCatalogOrSchema()